torchaudio/prototype/models/conformer_wav2vec2.py

```diff
@@ -0,0 +1,290 @@
+from typing import List, Optional, Tuple, Union
+
+from torch import nn, Tensor
+from torch.nn import Module
+from torchaudio.models import Wav2Vec2Model
+from torchaudio.models.conformer import ConformerLayer
+from torchaudio.models.rnnt import _TimeReduction
+from torchaudio.models.wav2vec2 import components
+
+
+class TimeReduction(Module):
+    """Extract features from input. Consists of time reduction and linear layer.
+
+    Args:
+        stride (int): number of frames to merge for the output frame
+        input_dim (int): input dimension of the tensor
+        output_dim (int): output dimension of the tensor
+    """
+
+    def __init__(self, input_dim: int, output_dim: int, stride: int):
+        super().__init__()
+        self.time_reduction_layer = _TimeReduction(stride=stride)
+        self.linear_layer = nn.Linear(input_dim * stride, output_dim)
+
+    def forward(
+        self,
+        x: Tensor,
+        lengths: Optional[Tensor],
+    ) -> Tuple[Tensor, Optional[Tensor]]:
+        """
+        Args:
+            x (Tensor):
+                Input Tensor representing log Mel Spectrogram output. shape ``(batch, frame, dim)``.
+            length (Tensor or None, optional):
+                Valid length of each input sample. shape: ``(batch, )``.
+        """
+        # TODO: case if length is none?
```

Comment:
What is supposed to happen if length is None?

Reply:
supplying some dummy tensor to `self.time_reduction_layer` should suffice

Reply:
maybe initialize a lengths tensor with shape `(batch,)`, and every value is the second dimension of x.